### PR TITLE
Ep/qmml

### PR DESCRIPTION
**Pull request by epens94**
**Contains changes to work with QCML Dataset**

- First Draft of QCML Dataclass
- Inclusion of adaptive loss function
- Inclusion of Bernstein and PhysNet radial basis
- New Splitting class based on atom type

### CODE REVIEW

File: .vscode/extensions.json
Comment (stefaanhessmann) [inline .vscode/extensions.json:1]: I think this should be ignored

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1 @@
+include src/schnetpack/train/ressources/partition_spline_for_robust_loss.npz
```
Comment (stefaanhessmann) [inline MANIFEST.in:1]: What is this?

File: src/schnetpack/configs/experiment/qcml.yaml
Comment (stefaanhessmann) [inline src/schnetpack/configs/experiment/qcml.yaml:1]: I would keep this private/local and not upload to schnetpack

File: src/schnetpack/configs/data/qcml.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+defaults:
+  - custom
+
+_target_: schnetpack.datasets.QCML
+
+datapath: ${run.data_dir}/qcml.db  # data_dir is specified in train.yaml
+batch_size: 50
+num_train: 0.90
+num_val: 0.05
+load_properties: [formation_energy,forces,charge,multiplicity]
```
Comment (stefaanhessmann) [inline src/schnetpack/configs/data/qcml.yaml:10]: I don`t think it is possible to pass a list like this. If I remember correctly it should work like

```
load properties:
    - formation_energy
    - forces
    - ...
```

File: src/schnetpack/configs/model/representation/nuclear_embedding/easy.yaml
Comment (stefaanhessmann) [inline src/schnetpack/configs/model/representation/nuclear_embedding/easy.yaml:1]: remove

File: pyproject.toml [hunk 3]
```diff
@@ -57,5 +57,8 @@ script-files = [
     "src/scripts/spkdeploy",
 ]
 
+# Ensure package data such as resources are included
+package-data = { "schnetpack.train" = ["ressources/partition_spline_for_robust_loss.npz"] }
```
Comment (stefaanhessmann) [inline pyproject.toml:61]: same here

File: src/schnetpack/configs/model/representation/nuclear_embedding/complex.yaml
Comment (stefaanhessmann) [inline src/schnetpack/configs/model/representation/nuclear_embedding/complex.yaml:1]: I would maybe directly add this in the experiment yaml file and not as a separate config file, because it is barely used at the moment

File: src/schnetpack/configs/model/representation/painn.yaml [hunk 4]
```diff
@@ -1,5 +1,6 @@
 defaults:
   - radial_basis: gaussian
+  - nuclear_embedding: complex
```
Comment (stefaanhessmann) [inline src/schnetpack/configs/model/representation/painn.yaml:3]: remove this and leave default value as it is

File: src/schnetpack/configs/task/scheduler/multistep.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# @package task
+scheduler_cls: torch.optim.lr_scheduler.MultiStepLR
+scheduler_monitor: val_loss
+scheduler_args:
+  milestones: [3,9,15,18,24,30,36]
```
Comment (stefaanhessmann) [inline src/schnetpack/configs/task/scheduler/multistep.yaml:5]: the milestones are very specific to the experiment. I would either set them to `???` or just define the scheduler in the experiment.yaml

File: src/schnetpack/data/atoms.py [hunk 6]
```diff
@@ -344,8 +344,10 @@ def _get_properties(
         properties = {}
         properties[structure.idx] = torch.tensor([idx])
         for pname in load_properties:
+
```
Comment (stefaanhessmann) [inline src/schnetpack/data/atoms.py:347]: remove blank lines

File: MANIFEST.in (hunk 1 above)
Comment (epens94) [reply MANIFEST.in:1]: From this file values are loaded which are used to approximate the partition function to for the adaptive loss fn

File: src/schnetpack/configs/experiment/qcml.yaml
Comment (epens94) [reply src/schnetpack/configs/experiment/qcml.yaml:1]: Do you mean because of the data ? or the attached comments ?

File: pyproject.toml (hunk 3 above)
Comment (epens94) [reply pyproject.toml:61]: this is needed to include the file when building the package

File: src/schnetpack/configs/model/representation/painn.yaml (hunk 4 above)
Comment (epens94) [reply src/schnetpack/configs/model/representation/painn.yaml]: I agree, done in 0f6fc8bae9a950c8f6130b92eb013b4d6f01c95f

File: src/schnetpack/configs/task/scheduler/multistep.yaml (hunk 5 above)
Comment (epens94) [reply src/schnetpack/configs/task/scheduler/multistep.yaml]: I agree, done in ca4acd7070786631797886a25488d19794c7e214

File: src/schnetpack/data/atoms.py (hunk 6 above)
Comment (epens94) [reply src/schnetpack/data/atoms.py]: done
